ballerina/tests/to_xml_test.bal

```diff
@@ -0,0 +1,852 @@
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testMapJsonToXml1() returns error? {
+    map<json> data = {id: 30, customer: {name: "Asha", age: 10}};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<root><id>30</id><customer><name>Asha</name><age>10</age></customer></root>`,
+                    msg = "testMapJsonToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testDefaultRecordToXml1() returns error? {
+    record {int id;} data = {id: 30};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<id>30</id>`, msg = "testDefaultRecordToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testDefaultRecordToXml2() returns error? {
+    record {int id; string name;} data = {id: 30, name: "Asha"};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<root><id>30</id><name>Asha</name></root>`,
+                      msg = "testDefaultRecordToXml2 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testMapStringToXml1() returns error? {
+    map<string> data = {"id": "30"};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<id>30</id>`, msg = "testMapStringToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testMapStringToXml2() returns error? {
+    map<string> data = {"id": "30", "name": "Asha"};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<root><id>30</id><name>Asha</name></root>`,
+                      msg = "testMapStringToXml2 result incorrect");
+}
+
+@Name {
+    value: "Customers"
+}
+@Namespace {
+    prefix: "ns",
+    uri: "http://sdf.com"
+}
+type Customer record {
+
+    @Name {
+        value: "employeeName"
+    }
+    @Attribute
+    string ns\:name;
+
+    int age;
+};
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testRecordWithAnnotationToXml1() returns error? {
+    Customer data = {ns\:name: "Asha", age: 10};
+    xml result = check toXml(data);
+    test:assertEquals(result,
+                    xml `<ns:Customers xmlns:ns="http://sdf.com" ns:employeeName="Asha"><age>10</age></ns:Customers>`,
+                    msg = "testRecordWithAnnotationToXml1 result incorrect");
+}
+
+@Namespace {
+    prefix: "ns",
+    uri: "http://sdf.com"
+}
+@Name {
+    value: "Customers"
+}
+type Customer2 record {
+
+    @Name {
+        value: "employeeName"
```

Comment:
Don't we need to add prefix here `ns:employeeName`

Reply:
This is my mistake. Yes, we need to add.